Keas.Mvc/ClientApp/components/Keys/AssignKeySerial.tsx

```diff
@@ -46,18 +46,20 @@ export default class AssignKey extends React.Component<IProps, IState> {
   constructor(props: IProps) {
     super(props);
 
-    const date = (!!this.props.selectedKeySerial && !!this.props.selectedKeySerial.assignment) 
-      ? moment(this.props.selectedKeySerial.assignment.expiresAt)
+    const assignment = props.selectedKeySerial && props.selectedKeySerial.keySerialAssignment;
+
+    const date = (!!assignment) 
```

Comment:
what's the point of the double bang given we know from the line above this is boolean

Reply:
It's not a bool.
The && is null coalesce.

Reply:
still not seeing the point of it

Reply:
`props.selectedKeySerial` can be null and the next several lines are a lot cleaner with the `assignment` instead of `props.selectedKeySerial.keySerialAssignment`.